### ISSUE TITLE
Missing icons for menu items on Gnome 42

### ISSUE THREAD

**Issue by PJansky** [xournalpp/xournalpp#3945]
Icons for menu items in the menubar are missing (e.g. "Open" or "Save") on Gnome 42, presumably due to switch to GTK4 and libadwaita.

The icons are completely missing on Xorg, Wayland displays a missing icon symbol.

**Affects versions :**
 - OS: Arch Linux
 - (Linux only) Desktop environment: Gnome-Wayland, Gnome-Xorg
 - Version of libgtk: 3.24.33
 - Version of Xournal++: 1.1.1+dev
 - Installation method: native build from AUR (xournalpp-git)

**To Reproduce**
1. Open "File" or any other menubar entry with icons next to menu items

**Screenshots of Problem**
Wayland:
![Wayland](https://user-images.githubusercontent.com/46072333/162189007-c68c9fe8-d8a7-4fec-90fe-3923945c6315.png)
Xorg:
![Xorg](https://user-images.githubusercontent.com/46072333/162189009-a6d3ffbd-8993-4cc8-98aa-20b3765234d1.png)

**Comment by rolandlo** [comment]
For me (on Ubuntu 22.04) it looks like for you on Xorg, no matter if I run Xournal++ in a Wayland or X11 session.
Did you really see icons there before switching?

**Comment by PJansky** [comment]
Yes, there should be icons there. Here is an example with KDE Plasma. In Gnome the icons were also there (the obviously just looked a bit different)
![KDE](https://user-images.githubusercontent.com/46072333/162275826-7a3f051c-6a46-486b-bd30-161a1fd465f9.png)

**Comment by K4LCIFER** [comment]
Gnome 42 Wayland, and Xournal++ Installed with Flatpak.
![image](https://user-images.githubusercontent.com/7829231/162300414-5ebc6bfe-2fe1-4dfc-9bf5-fcb59b5feeb6.png)

**Comment by PJansky** [comment]
To make shure I'm not going insane, I spun up a VM with Gnome 41.5. The icons are clearly there. @K4LCIFER it's good to know, that the flatpak version is also affected.
Gnome 41.5
![Gnome 41-5](https://user-images.githubusercontent.com/46072333/162402043-6fefeafd-5afc-4bc4-8e13-e377f5f070b7.png)

**Comment by rolandlo** [comment]
What is used for these menu icons are `GtkImageMenuItem`s. Those have been deprecated since Gtk 3.10 according to
https://docs.gtk.org/gtk3/class.ImageMenuItem.html . They may not be functional any more in the latest Gtk version(s). The documentation also says: `Note that the user may disable display of menu icons using the GtkSettings:gtk-menu-images setting`.
Maybe we should just follow the recommendations and replace those `GtkImageMenuItem`s with ordinary `GtkMenuItems`s (that don't show icons).

**Comment by langfingaz** [comment]
> Maybe we should just follow the recommendations and replace those `GtkImageMenuItem`s with ordinary `GtkMenuItems`s (that don't show icons).

Switching to `GtkMenuItem`s sounds good. But even with those one can add icons as stated on your link to the documentation:

> If you want to display an icon in a menu item, you should use GtkMenuItem and pack a GtkBox with a GtkImage and a GtkLabel instead.

I myself liked the icons in that menu. But we can discuss if we want to keep them. I think they help non technical users to find the save as or print items quicker.

**Comment by couldNOTcompute** [comment]
**Affects versions :**
    OS: Windows 10 Pro
    libGTK: 3.24.33
    Version of Xournal++: 1.1.1+dev (dc5188ab-dirty)
    Installation method: Installer

**Description:**
There are Icons missing for buttons in the sidebar.

![xpp_missing-icons](https://user-images.githubusercontent.com/95862384/167440287-b445db10-c1e4-46a9-806b-4af5b8542d6e.PNG)
(From the placeholder Icons it looks like the same issue)

**Comment by rolandlo** [comment]
@couldNOTcompute The missing icons are the Gtk stock icons `gtk-close`, `gtk-go-up`, `gtk-go-down`, `gtk-copy` and `gtk-delete`. Looks like they are contained in the Humanity theme, but not in the Adwaita theme, which gets packaged under Windows. We'll have to fix that.

**Comment by mjg** [comment]
I'm not really sure what's happening here and if that's related - could it be that they got renamed? I see "symbolic" versions in Adwaita of those which are reported missing. In fact, Xpp coude seems to refere to some other icons in their symbolic variant (?) if that's what is. Although things like this made me suspect a bug in the icon theme:
```
/usr/share/icons/Adwaita/64x64/actions/go-down-symbolic.symbolic.png
/usr/share/icons/Adwaita/96x96/actions/go-down-symbolic.symbolic.png
/usr/share/icons/Adwaita/scalable/actions/go-down-symbolic.svg
```

**Comment by All3xJ** [comment]
> **Affects versions :** OS: Windows 10 Pro libGTK: 3.24.33 Version of Xournal++: 1.1.1+dev ([dc5188a](https://github.com/xournalpp/xournalpp/commit/dc5188abc34604430d8af37f65847e36e1e4eb10)-dirty) Installation method: Installer
> 
> **Description:** There are Icons missing for buttons in the sidebar.
> 
> ![xpp_missing-icons](https://user-images.githubusercontent.com/95862384/167440287-b445db10-c1e4-46a9-806b-4af5b8542d6e.PNG) (From the placeholder Icons it looks like the same issue)


Some issue under GNOME in Archlinux

**Comment by rolandlo** [comment]
This issue is probably coming from https://gitlab.gnome.org/GNOME/gtk/-/issues/4754

**Comment by SirAutism** [comment]
> 

I am having this issue aswell on windows, has it been fixed for you?

**Comment by SubhamSatyajeet** [comment]
I'm able to see this issue in Xournal++ 1.1.2, Windows 10 22H2. If anyone knows a workaround to get the missing icons to show, please tell.

**Comment by All3xJ** [comment]
issue still here still unfixed. I have version 1.1.2+dev , Gnome 43.1, Archlinux

**Comment by o-kotb** [comment]
It looked like this initially after installing either flathub or fedora flatpak. But all the icons came back after rebooting my system.

**Comment by bhennion** [comment]
> This issue is probably coming from https://gitlab.gnome.org/GNOME/gtk/-/issues/4754

Closing as the upstream issue has been solved 10 months ago (GTK 3.24.34 or more recent should contain the fix).

**Comment by rolandlo** [comment]
This was fixed in https://github.com/xournalpp/xournalpp/pull/4544
